aws/resource_aws_cloudhsm2_hsm.go

```diff
@@ -0,0 +1,285 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudhsmv2"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func resourceAwsCloudHsm2Hsm() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudHsm2HsmCreate,
+		Read:   resourceAwsCloudHsm2HsmRead,
+		Update: resourceAwsCloudHsm2HsmUpdate,
+		Delete: resourceAwsCloudHsm2HsmDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceAwsCloudHsm2HsmImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(120 * time.Minute),
+			Update: schema.DefaultTimeout(120 * time.Minute),
+			Delete: schema.DefaultTimeout(120 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"ip_address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"hsm_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hsm_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"hsm_eni_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudHsm2HsmImport(
+	d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	d.Set("hsm_id", d.Id())
+	return []*schema.ResourceData{d}, nil
+}
+
+func describeHsm(d *schema.ResourceData, meta interface{}) (*cloudhsmv2.Hsm, error) {
+	conn := meta.(*AWSClient).cloudhsmv2conn
+
+	clusterId := d.Get("cluster_id").(string)
+	hsmId := d.Id()
+	filters := []*string{&clusterId}
+	result := int64(1)
+	out, err := conn.DescribeClusters(&cloudhsmv2.DescribeClustersInput{
```

Comment:
Instead of reimplementing most of the logic here, should this function just use `describeCloudHsm2Cluster`?